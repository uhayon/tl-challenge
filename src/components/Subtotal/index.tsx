import { FC } from 'react';
import { useDiscounts } from '../../hooks/useDiscounts';
import { useProducts } from '../../hooks/useProducts';
import styles from './subtotal.module.scss';

const Subtotal: FC = () => {
  const { selectedItems } = useProducts();
  const { currentTier } = useDiscounts();
  const subtotal = selectedItems.reduce(
    (total, { price, count }) => total + price * count,
    0
  );

  return (
    <div className={styles.subtotal}>
      <div className={styles.categoryContainer}>
        <span className={styles.label}>Build Your Kit Subtotal</span>
        <span className={styles.value}>{`$${subtotal}`}</span>
      </div>
      {currentTier && (
        <div className={styles.categoryContainer}>
          <div>
            <span className={styles.label}>Discount</span>
            <span
              className={styles.tierDescription}
            >{`(Spend $${currentTier.basePrice}, Save $${currentTier.discount})`}</span>
          </div>
          <span
            className={`${styles.value} ${styles.discount}`}
          >{`-$${currentTier.discount}`}</span>
        </div>
      )}
    </div>
  );
};

export default Subtotal;
