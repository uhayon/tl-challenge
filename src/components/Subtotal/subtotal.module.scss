.subtotal {
	display: flex;
	flex-direction: column;

	margin-top: 30px;

	.categoryContainer {
		display: flex;
		justify-content: space-between;

		margin: 2px 0;

		.label {
			font-size: 14px;
			line-height: 18px;

			letter-spacing: -0.01em;
		}

		.tierDescription {
			margin-left: 2px;

			font-size: 12px;
			line-height: 14px;
			color: #666;
		}

		.value {
			font-size: 14px;
			font-weight: bold;
			line-height: 18px;

			&.discount {
				color: #c51e1e;
			}
		}
	}
}
