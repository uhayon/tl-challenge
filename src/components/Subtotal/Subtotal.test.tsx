import { render, screen } from '@testing-library/react';
import Subtotal from './';

describe('Subtotal', () => {
  it('Should render successfully', () => {
    const { container } = render(<Subtotal />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  describe('Without Discount', () => {
    beforeAll(() => {
      jest.mock('../../hooks/useDiscounts', () => ({
        useDiscounts: () => ({
          currentTier: null,
        }),
      }));
      jest.resetAllMocks();
      jest.restoreAllMocks();
      jest.resetModuleRegistry();
      jest.resetModules();
    });
    it('Should not render', () => {
      render(<Subtotal />);
      const discountElement = screen.queryByText('Discount');
      expect(discountElement).toBeNull();
    });
  });
});
