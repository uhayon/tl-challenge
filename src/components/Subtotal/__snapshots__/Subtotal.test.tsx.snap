// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtotal Should render successfully 1`] = `
<div>
  <div
    class="subtotal"
  >
    <div
      class="categoryContainer"
    >
      <span
        class="label"
      >
        Build Your Kit Subtotal
      </span>
      <span
        class="value"
      >
        $0
      </span>
    </div>
  </div>
</div>
`;
